### ISSUE TITLE
add header with provider information

### ISSUE THREAD

**Issue by pelikhan** [promptfoo/promptfoo#710]
We have a test scenario with multiple providers and a few tests. Would it be possible to add a header cell on the output column with the id/label of the provider?

**Comment by typpo** [comment]
Hi @pelikhan, I may not be understanding this ask correctly.  Currently the id/label should appear at the top of the table in the header.

![image](https://github.com/promptfoo/promptfoo/assets/310310/6c4e4151-8aba-480b-8eb2-07b9fc362ded)

**Comment by pelikhan** [comment]
Ah, I don't see it.
![image](https://github.com/promptfoo/promptfoo/assets/4175913/c00b1443-104e-4714-a626-77ed8114a283)

Could this be because I use a custom provider?

```yaml
description: summarize all files
prompts:
  - summarize
providers:
  - id: provider.mjs
    label: gpt-4
    config:
      model: gpt-4
tests:
  - vars:
      files: src/rag/markdown.md
    asserts:
      - type: llm-rubric
        value: is a summary
      - type: factuality
        value: Markdown is a text-based syntax to generate documents
      - type: icontains
        value: Markdown
```

**Comment by pelikhan** [comment]
(accidental closing)

**Comment by typpo** [comment]
Ah, this is happening because the provider label is intentionally hidden when all providers are the same in the eval.  Potentially could remove this behavior if people find it confusing or inconsistent https://github.com/promptfoo/promptfoo/blob/main/src/web/nextui/src/app/eval/ResultsTable.tsx#L756-L758

**Comment by pelikhan** [comment]
Great, that makes sense.

The provider is always the same JavaScript file, the interesting one is "label".

(Is label something I hallucinated?) It could de displayed on the UI of present always.

**Comment by typpo** [comment]
Just made #721 - I think we can just show the provider always, for consistency :)  Thanks for flagging this

**Comment by typpo** [comment]
This change is included in release [0.56.0](https://github.com/promptfoo/promptfoo/releases/tag/0.56.0)

**Comment by pelikhan** [comment]
We also need a way to display a label instead of the provider; since the provider id is always going to be the file name (provider.mjs in this case)

![image](https://github.com/promptfoo/promptfoo/assets/4175913/03c3ca8b-6e1e-4290-9146-c6d8c5e45f85)
